src/ResultSet.php

```diff
@@ -48,4 +48,27 @@ public function getIterator(): ArrayIterator
     {
         return new ArrayIterator($this->results);
     }
+
+    public function getErrors(): self
+    {
+        $resultSet = new self();
+        foreach ($this->results as $attribute => $result) {
+            if (!$result->isValid()) {
+                $resultSet->addResult($attribute, $result);
+            }
+        }
+
+        return $resultSet;
+    }
+
+    public function isValid(): bool
+    {
+        foreach ($this->results as $result) {
+            if (!$result->isValid()) {
```

Comment:
Please use `addResult` method to catching an added new errored result:
If an added result has errors then whole ResultSet will have an error.